lockfile/src/parsers/pypi.rs

```diff
@@ -52,10 +53,26 @@ fn line(input: &str) -> IResult<&str, &str> {
         line = "";
     }
 
+    // Ignore index config options.
+    //
+    // Since `ThirdPartyVersion` only allows a single registry, we only record the
+    // primary one.
+    if let Some(index_url) = line
+        .strip_prefix("--index-url")
+        .and_then(|line| line.strip_prefix(['=', ' ']))
```

Comment:
Just to make sure, I'd assume `--index-url = https://whatever.com` is **not** valid?

Reply:
Correct...that form is **not** valid:
<img width="1032" alt="image" src="https://github.com/phylum-dev/cli/assets/18729796/8c2e1312-266f-40f4-8d2d-4c9de7e61a2a">

Reply:
There are other weird edge cases... But I don't want to waste a bunch of time on them...

```
> cat req.txt
--index-url https://pypi.org/simple AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA

pyyaml

> pip install --dry-run -r req.txt
Collecting pyyaml
  Downloading PyYAML-6.0.1-cp310-cp310-macosx_11_0_arm64.whl (169 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 169.3/169.3 kB 2.1 MB/s eta 0:00:00
Would install PyYAML-6.0.1
```